applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -533,6 +551,42 @@ static void usb_wakeup(void)
 	}
 }
 
+static uint32_t get_report_bm(size_t hid_id)
+{
+	BUILD_ASSERT(REPORT_ID_COUNT <=
+		     sizeof(usb_hid_device[0].report_bm) * CHAR_BIT);
+#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
+	BUILD_ASSERT(ARRAY_SIZE(usb_hid_report_bm) ==
+		     CONFIG_USB_HID_DEVICE_COUNT);
+	return usb_hid_report_bm[hid_id];
+#else
+	return UINT32_MAX;
+#endif
+}
+
+static void verify_report_bm(void)
+{
+
+#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
```

Comment:
Why not:
```
#if defined(CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION) && defined(CONFIG_ASSERT)
```

Reply:
I noticed it too but personally I don't mind so much. This was ping-ponging long enough already and don't think it's worth another push and CI rerun.
In general we use compiler based checks but I asked Marek to use preprocessor to clean up file header. As a result both conditions should go to preprocessor check but as mentioned above - I think this review reached its limit of bikeshedding.

Reply:
fixed

Reply:
I agree, it probably wasn't worth the cycle, but I have seen quite a few of those before and this one stood out because it was combining `IS_ENABLED()` and `#ifdef`